src/playgroundrl_envs/games/petting_zoo.py

```diff
@@ -77,9 +86,18 @@ def submit_action(self, raw_action, player_sid=""):
         return True
 
     def get_state(self, player_sid="", player_id=0):
-        # TODO: see if we need to change this based on the player accessing it
-        pickled = codecs.encode(pickle.dumps(self.state), "base64").decode()
-        return pickled, float(self.state[1])
+        if self.self_training:
+            pickled = codecs.encode(pickle.dumps(self.state), "base64").decode()
+            return pickled, float(self.state[1])        
+        else: 
+            agent = self.players[self.player_ids[self.player_moving_index]].agent_name
+            observation = self.latest_json_action
+            
+            state_jsonable = self.env.action_space(agent=agent).to_jsonable([observation])[0]
+            state_str = json.dumps(state_jsonable)        
+            
+            return state_str, float(self.state[1])
```

Comment:
similar for here. if the model is playing (not in self_training) it should still get the pickled state